astropy/units/core.py

```diff
@@ -1409,6 +1406,46 @@ def __repr__(self):
                          [']'])
                 return '\n'.join(lines)
 
+        def _repr_html_(self):
+            """
+            Outputs a HTML table representation within Jupyter notebooks.
+            """
+            if len(self) == 0:
+                return "<p>{}</p>".format(self.NO_EQUIV_UNITS_MSG)
+            else:
+                # HTML tags to use to compose the table in HTML
+                blank_table = '<table style="width:100%">{}</table>'
+                blank_row_container = "<tr>{}</tr>"
+                heading_row_content = "<th>{}</th>" * self.ROW_LEN
+                data_row_content = "<td>{}</td>" * self.ROW_LEN
+
+                # The HTML will be rendered & the table is simple, so don't
+                # bother to include newlines & indentation for the HTML code.
+                heading_row = blank_row_container.format(
+                    heading_row_content.format(*self.HEADING_NAMES))
+                data_rows = self.process_equivalent_units(self)
+                all_rows = heading_row
+                for row in data_rows:
+                    html_row = blank_row_container.format(
+                        data_row_content.format(*row))
+                    all_rows += html_row
+                return blank_table.format(all_rows)
+
+        @staticmethod
+        def process_equivalent_units(equiv_units_data):
```

Comment:
This should probably be a private method (``_process_equivalent_units``)

Reply:
Sure, I'll get it changed as such in my next commit. Thanks for the feedback.